packages/features/ee/billing/api/webhook/__handler.ts

```diff
@@ -0,0 +1,61 @@
+/// <reference types="stripe-event-types" />
+import { buffer } from "micro";
+import type { NextApiRequest } from "next";
+import type Stripe from "stripe";
+
+import stripe from "@calcom/app-store/stripepayment/lib/server";
+import { HttpError } from "@calcom/lib/http-error";
+
+/** Stripe Webhook Handler Mappings */
+export type SWHMap = {
+  [T in Stripe.DiscriminatedEvent as T["type"]]: {
+    [K in keyof T as Exclude<K, "type">]: T[K];
+  };
+};
+
+export type LazyModule<D> = Promise<{
+  default: (data: D) => unknown | Promise<unknown>;
+}>;
+
+type SWHandlers = {
+  [K in keyof SWHMap]?: LazyModule<SWHMap[K]["data"]>;
+};
+
+/** Just a shorthand for HttpError  */
+export class HttpCode extends HttpError {
+  constructor(statusCode: number, message: string) {
+    super({ statusCode, message });
+  }
+}
+
+/**
+ * Allows us to split big API handlers by webhook event, and lazy load them.
+ * It already handles the Stripe signature verification and event construction.
+ * @param handlers - A mapping of Stripe webhook event types to lazy loaded handlers.
+ * @returns A function that can be used as a Next.js API handler.
+ * @example
+ * ```ts
+ * stripeWebhookHandler({
+ *   "payment_intent.succeeded": import("./_lazyLoadedSuccessHandler"),
+ *   "customer.subscription.deleted": import("./_customer.subscription.deleted"),
+ * })
+ * ```
+ */
+export const stripeWebhookHandler = (handlers: SWHandlers) => async (req: NextApiRequest) => {
+  const STRIPE_WEBHOOK_SECRET = process.env.STRIPE_WEBHOOK_SECRET;
```

Comment:
This has to live in scope of the function as we arent able to change the value when mocking as this constant is outside of this fn scope.